src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Check.java

```diff
@@ -5213,6 +5213,18 @@ public Void visitTypeAsEnum(TypeElement e,
                             }
                         }
                     }
+
+                    case ENUM_CONSTANT -> {
+                        var field = (VarSymbol)enclosed;
+                        if (field.isEnum()) {
```

Comment:
It's not wrong, but is this check redundant?  When can an `ENUM_CONSTANT` not be `isEnum()` ?

Reply:
Yes, sorry for that, I've originally had the code elsewhere. The `isEnum` check should not be needed here.

Reply:
Code updated; all langtools tests pas when run locally. Thanks.